src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5812,10 +5814,22 @@ private static BoundExpression CreatePlaceholderIfNecessary(BoundExpression expr
                     }
 
                     VisitExpressionWithoutStackGuardEpilogue(receiver); // VisitExpressionWithoutStackGuard does this after visiting each node
-                    VisitRvalueEpilogue(receiver); // VisitRvalue does this after visiting each node
 
-                    receiverType = ResultType;
-                    CheckCallReceiver(receiver, receiverType, node.Method);
+                    // Only instance receivers go through VisitRvalue; arguments go through Visit.
+                    if (node.ReceiverOpt is not null)
+                    {
+                        VisitRvalueEpilogue(receiver); // VisitRvalue does this after visiting each node
+                        receiverType = ResultType;
+                        CheckCallReceiver(receiver, receiverType, node.Method);
+                        extensionReceiverResult = null;
+                    }
+                    else
+                    {
+                        Debug.Assert(node.InvokedAsExtensionMethod);
+                        var refKind = GetRefKind(node.ArgumentRefKindsOpt, 0);
+                        var annotations = GetCorrespondingParameter(0, node.Method.Parameters, node.ArgsToParamsOpt, node.Expanded).Annotations;
+                        extensionReceiverResult = VisitArgumentEvaluateEpilogue(receiver, default, refKind, annotations);
```

Comment:
Should we reset value of ```receiverType``` in some way? It is going to be used.

Reply:
Yes, good catch, thanks.